src/helpers/getShowUtils.ts

```diff
@@ -0,0 +1,63 @@
+import { ShowData, TvShowDetailsData, TvShowData, ShowProviders } from '../types/tmdb';
+
+/**
+ * This function is ran after the user enters a name of a TV Show.
+ * @returns {Promise<TvShowData>} | List of shows by searched query.
+ */
+const getShowsByName = async (name: string): Promise<TvShowData> => {
+    const response = await fetch(
+        `https://api.themoviedb.org/3/search/tv?api_key=${
+            import.meta.env.VITE_MOVIEDB_KEY
+        }&language=en-US&query=${name}&page=1&include_adult=false`
+    );
+    return response.json() as Promise<TvShowData>;
+};
+
+/**
+ * This function is ran for specific <ShowCard /> data with a TV Show ID.
+ * @returns {Promise<ShowData>} | Specific data for a TV Show that is not originally supplied by getMoviesByName.
+ */
+const getShowDetails = async (id: number): Promise<ShowData> => {
+    const response = await fetch(
+        `https://api.themoviedb.org/3/tv/${id}?api_key=${
+            import.meta.env.VITE_MOVIEDB_KEY
+        }&append_to_response=images,release_dates,content_ratings`
+    );
+    const returnRating = async (arr: TvShowDetailsData) => {
+        for (let i = 0; i < arr.content_ratings.results.length; i++) {
+            if (arr.content_ratings.results[i].iso_3166_1 === 'US') {
+                return arr.content_ratings.results[i].rating;
+            }
+        }
+        return null;
```

Comment:
We could return "No rating available" at this point so we still return a string and have something to display. Otherwise we probably want to conditionally render the text component that displays this info.